daemon/mgr/container.go

```diff
@@ -309,10 +309,11 @@ func (mgr *ContainerManager) Create(ctx context.Context, name string, config *ty
 	}
 
 	// set default log driver and validate for logger driver
-	if config.HostConfig.LogConfig == nil {
-		config.HostConfig.LogConfig = &types.LogConfig{
-			LogDriver: types.LogConfigLogDriverJSONFile,
-		}
+	// note: request from swarm is not standard. config.HostConfig.LogConfig is not nil, but its value is empty.
+	//       We need to be compatible with this legacy system.
+	if config.HostConfig.LogConfig == nil || config.HostConfig.LogConfig.LogDriver == "" {
```

Comment:
Could we use helper function to check the `config.HostConfig.LogConfig` is empty or not? 

Like, 

```
func isEmptyLogConfig(xx *types.LogConfig) bool {
      if config.HostConfig.LogConfig == nil  {
           return true
      }
      return config.HostConfig.LogConfig.LogDriver == "" && len(config.HostConfig.LogConfig.LogOpts) == 0
}
```

WDTY?

Reply:
Yep. Thanks